apps/api/src/domain/models/resource.entity.ts

```diff
@@ -78,6 +79,15 @@ export abstract class Resource extends Aggregate {
         } as unknown as DeepPartial<DTO<T>>);
     }
 
+    @UpdateMethod()
+    delete<T extends Resource>(this: T): ResultOrError<T> {
+        if (!this.delete) return new ResourceDeletedError(this.getCompositeIdentifier());
```

Comment:
Let's call this property `hasBeenDeleted`. I'd go with `isArchived`, but I'm worried we may add an 'archived' state that has a different meaning later on.

Also, be sure to declare the `hasBeenDeleted: boolean` field above

Reply:
done